openpifpaf/network/nets.py

```diff
@@ -188,7 +189,8 @@ def model_migration(net_cpu):
             hn.meta.base_stride = net_cpu.base_net.stride
         if hn.meta.head_index is None:
             hn.meta.head_index = hn_i
-
+        if hn.meta.name == 'cif' and hn.meta.score_weights is None:
```

Comment:
I think there are two cases here that need to be treated separately:
1. The head meta does not have a `score_weights` attribute, i.e. when you load a checkpoint. Actually, my current expectation is that you cannot load a checkpoint. And this line raises an exception. Use `hasattr()` here.
2. The `score_weights` attribute exists and is None. That is actually the default and therefore should not be modified. You are already dealing with None inside the constructor of Annotation.

Reply:
Ok, I was wrong. This works because `score_weights` is also a class attribute. An instance loaded from a checkpoint won't have the instance attribute but it will resolve to the class attribute with the same name. To test for the presence of the instance attribute, you can do `'score_weights' in vars(hn.meta)`.

Reply:
Hi @svenkreiss, many thanks for your revision.

To summarise,  `score weights` is defined in two places. In the class `Cif` (headmeta.py) it is defined as a class attribute as None. In the class `Annotation` (in annotations.py) it is defined as an instance attribute: None if nothing else is provided.

When running predictions, in the migration functions `Annotation` has not yet been instantiated but `score_weights` exists as class attribute. Later on, in the decoder, `Annotation` is instantiated and then `score_weights` becomes an instance attribute, either None or custom values. 
By using `vars()`, we only check for the presence of instance attributes and this allows us to distinguish between old models (without an instance attribute) and new models (instance attribute None or custom). 
This should also allow loading checkpoints for training as well.